File: Shared/MapperResolvers/AuthorResolver.cs
using AutoMapper;
using EntityModels;

namespace Shared.MapperResolvers
{
    public class AuthorResolver : ITypeConverter<AuthorMappingObjectEM, AuthorEM>
    {
        public AuthorEM Convert(AuthorMappingObjectEM source, AuthorEM destination, ResolutionContext context)
        {
            if (source.Author == null)
            {
                return null;
            }
            else
            {
                destination = source.Author;
                destination.Books = source.Books;
                return destination;
            }
        }
    }
}

File: SharedVB/MapperResolvers/AuthorResolver.vb
Imports AutoMapper
Imports EntityModels

Namespace Resolvers
    Public Class AuthorResolver
        Implements ITypeConverter(Of AuthorMappingObjectEM, AuthorEM)

        Public Function Convert(source As AuthorMappingObjectEM, destination As AuthorEM, context As ResolutionContext) As AuthorEM Implements ITypeConverter(Of AuthorMappingObjectEM, AuthorEM).Convert
            If source.Author Is Nothing Then
                Return Nothing
            Else
                destination = source.Author
                destination.Books = source.Books
                Return destination
            End If
        End Function
    End Class
End Namespace

